### PR TITLE
Xrt 2947: Add XRT Library Example

### PR DESCRIPTION
**Pull request by JonathanWhiteUK**

### CODE REVIEW

File: Library/readme.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
+# Using XRT as a Library
+
```
Comment (Tylerg2471) [inline Library/readme.md]: * Rename readme.md to README.md
* Need to add in where/how to install required packages (Link to docs would be best)
* Add a link to API docs for Xrt as a library

File: Library/demo.c [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+{
+  iot_container_stop (container);
+}
+
+static inline void free_xrt(iot_container_t *container)
+{
+  iot_container_free (container);
+}
+
+struct bus_component_t
```
Comment (Tylerg2471) [inline Library/demo.c]: typedef this

File: Library/demo.c [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+static void signal_handler (int sig)
+{
+  (void)sig;
+  atomic_store (&stopped, true);
+  pthread_cond_signal (&cond);
+}
+
+static inline iot_container_t* init_xrt (char *config_uri)
+{
+  struct sigaction signal_action = {0};
```
Comment (Tylerg2471) [inline Library/demo.c:40]: Need a null check for config_uri?

File: Library/demo.c [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+      iot_data_vector_iter_t iter;
+      iot_data_vector_iter (device_list, &iter);
+      while (iot_data_vector_iter_next (&iter))
+      {
+	      const iot_data_t *ele_data = iot_data_vector_iter_value (&iter);
+      	const iot_data_t *device_get =  create_get_message (ele_data);
+      	iot_log_info (bus->logger, "request sent: %s %d", iot_data_to_json (device_get), iot_data_type (device_get));
+      	xrt_bus_pub_push (bus->pub,iot_data_add_ref (data),true); // list message gets freed on publish
+
+      	if (wait_for_reply(100))
```
Comment (Tylerg2471) [inline Library/demo.c]: Why hardcoded 100? Could be a define

File: Library/demo.c [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+    	  {	
+          iot_log_info (bus->logger, "received reply: %s", iot_data_to_json (returned));
+      	}
+      	else
+	      {
+	        iot_log_info (bus->logger, "did not receive reply");
+	      }
+	      sleep(5);
```
Comment (Tylerg2471) [inline Library/demo.c]: Should we be using a sleep? If so would be better to define the value rather than magic number

File: Library/demo.c [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+      iot_log_info (bus->logger, "did not receive reply");
+    }
+    iot_data_free (returned);
+  }
+  stop_bus (bus);
+  stop_xrt (container);
+
+  free_xrt (container);
+
+  return 0;
```
Comment (Tylerg2471) [inline Library/demo.c:211]: Should this always return 0? Could return other value for error

File: Library/demo.c [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+      	else
+	      {
+	        iot_log_info (bus->logger, "did not receive reply");
+	      }
+	      sleep(5);
+      }
+    }
+    else
+    {
+      iot_log_info (bus->logger, "did not receive reply");
```
Comment (Tylerg2471) [inline Library/demo.c]: Think messages like this should be iot_log_error or warn rather than info.

File: Library/demo.c [hunk 8]
```diff
@@ -0,0 +1,200 @@
+#include <signal.h>
```
Comment (Tylerg2471) [inline Library/demo.c:6]: Couple things not line specific:
* Does the c and cpp files need copyright at the top?
* Formatting needs fixed (to coding standards) 
* Not checking return values of functions that have return values

File: Library/demo.c [hunk 9]
```diff
@@ -0,0 +1,200 @@
...
+{
+  (void)sig;
+  atomic_store (&stopped, true);
+  pthread_cond_signal (&cond);
+}
+
+static inline iot_container_t* init_xrt (char *config_uri)
+{
+  struct sigaction signal_action = {0};
+  memset (&signal_action, 0, sizeof (signal_action));
```
Comment (Tylerg2471) [inline Library/demo.c]: Don't need to memset as initialising to 0

File: Library/demo.cpp [hunk 10]
```diff
@@ -0,0 +1,247 @@
+#include <signal.h>
```
Comment (Tylerg2471) [inline Library/demo.cpp:5]: Comments in .c file may carry over

**Review by Tylerg2471 — CHANGES_REQUESTED** [review]
Few things to look at

File: Library/demo.c (hunk 4 above)
Comment (JonathanWhiteUK) [reply Library/demo.c]: Done in C

File: Library/demo.c (hunk 3 above)
Comment (JonathanWhiteUK) [reply Library/demo.c:40]: Done in C

File: Library/demo.c (hunk 5 above)
Comment (JonathanWhiteUK) [reply Library/demo.c]: Done in C

File: Library/demo.c (hunk 7 above)
Comment (JonathanWhiteUK) [reply Library/demo.c]: Done in C

File: Library/demo.c (hunk 2 above)
Comment (JonathanWhiteUK) [reply Library/demo.c]: Done in C

File: Library/demo.c (hunk 3 above)
Comment (JonathanWhiteUK) [reply Library/demo.c:40]: Fixed in C++

File: Library/demo.c (hunk 4 above)
Comment (JonathanWhiteUK) [reply Library/demo.c]: Fixed in C++

File: Library/demo.c (hunk 5 above)
Comment (JonathanWhiteUK) [reply Library/demo.c]: Fixed in C++

File: Library/demo.c (hunk 7 above)
Comment (JonathanWhiteUK) [reply Library/demo.c]: Done in C++

**Review by Tylerg2471 — APPROVED** [review]
lgtm

**Comment by BinduRao2018** [comment]
- Update 'README' at the top level

**Pull request merged**